### PR TITLE
Update analyzer and Dart version to 2.17

### PR DESCRIPTION
**Pull request by Carapacik**
In my project I use freezed and also analyzer version 5(#499).  So I removed deprecated methods to their alternative.

- assertFinal -> declareFinal
- assertVar -> declareVar
- assertConst -> declareConst
- element -> element2
- enclosingElement -> enclosingElement3

Also, to use analyzer 4.6.0 and higher, need Dart 2.17

### CODE REVIEW

**Comment by Carapacik** [comment]
It can also, in addition to the generator, update the version and basically retrofit

**Comment by trevorwang** [comment]
> It can also, in addition to the generator, update the version and basically retrofit

Can you please update the retrofit accordingly?

**Comment by Carapacik** [comment]
Ok

**Comment by trevorwang** [comment]
Please also update the dart version to ensure the test action working.

https://github.com/trevorwang/retrofit.dart/blob/4334be1b9c49703aa55fd471248aabda6a6f1155/.github/workflows/test.yml#L14

**Comment by Carapacik** [comment]
> Please also update the dart version to ensure the test action working.
> 
> https://github.com/trevorwang/retrofit.dart/blob/4334be1b9c49703aa55fd471248aabda6a6f1155/.github/workflows/test.yml#L14

Update on 2.17?

**Comment by Carapacik** [comment]
Done

**Comment by Carapacik** [comment]
Strange, for some reason RequestOptions _setStreamType<T> began to be generated

**Comment by Carapacik** [comment]
A little later I will try to fix the tests and find errors

**Comment by Carapacik** [comment]
Everything works fine on the 2.16 version, strange

**Comment by trevorwang** [comment]
it seems to caused by generating code with new dart fmt

发自我的iPhone

**Comment by Carapacik** [comment]
@trevorwang All done and all test passed, but in pubspec.yaml:

> https://github.com/Carapacik/retrofit.dart/blob/8de2e4a253e9514dc8b27633fd346e2b2fcd5652/generator/pubspec.yaml#L15

**Comment by Carapacik** [comment]
It works for me in my project and all test passed.
I have set an upper bound for _code_builder_, so with the new analyzer and Dart 2.17 everything works ok.

**Comment by Carapacik** [comment]
@trevorwang Check this plz

>  All done and all test passed, but in pubspec.yaml:
> 
> ```yaml
>   code_builder: 4.2.0
> ```
> 
> > https://github.com/Carapacik/retrofit.dart/blob/8de2e4a253e9514dc8b27633fd346e2b2fcd5652/generator/pubspec.yaml#L15

File: generator/pubspec.yaml [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
 dependencies:
-  dio: ^4.0.1
-  source_gen: ^1.0.1
-  built_collection: ^5.0.0
-  code_builder: ^4.0.0
-  tuple: ^2.0.0
+  analyzer: '>=4.6.0 <6.0.0'
+  build: ^2.3.1
+  built_collection: ^5.1.1
+  code_builder: '>=4.0.0 <4.3.0'
```
Comment (Carapacik) [inline generator/pubspec.yaml:15]: This line

**Comment by Carapacik** [comment]
In short, all the changes:
- Update analyzer version to `'>=4.6.0 <6.0.0'` and replace:
  - assertFinal -> declareFinal
  - assertVar -> declareVar
  - assertConst -> declareConst
  - element -> element2
  - enclosingElement -> enclosingElement3
- Update code_builder version to  `'>=4.0.0 <4.3.0'`
- Update Dart to 2.17
- Update test workflow
- Fix typo:
  - sepcefied -> specified 
  - httpMehod  -> httpMethod 
  - wrapperedReturnType -> wrappedReturnType 
  - isWrappered -> isWrapped
  - innnerType  -> innerType 
  - optinalFile -> optionalFile 
- Extract 'value' to const _valueVar

File: generator/pubspec.yaml (hunk 1 above)
Comment (trevorwang) [reply generator/pubspec.yaml:15]: OK   

code_builder can be updated in future release.

**Comment by trevorwang** [comment]
fix #499

**Comment by trevorwang** [comment]
@Carapacik   generator 4.1.0+1 has been released to pub.dev  Thanks for your collabration.

**Comment by a1573595** [comment]
@trevorwang 

retrofit_generator: ^4.0.3+2 work fine with mockito: ^5.3.2, but retrofit_generator: ^4.1.0+1 has conflict.

#504

**Pull request merged**